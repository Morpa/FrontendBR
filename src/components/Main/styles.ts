import styled, { css } from 'styled-components'

import * as ButtonStyles from 'components/Button/styles'

export const Wrapper = styled.main`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    text-align: center;
    padding: ${theme.spacings.medium};
    align-items: center;
    justify-content: center;

    ${ButtonStyles.Wrapper} {
      margin-top: ${theme.spacings.medium};
    }
  `}
`

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
