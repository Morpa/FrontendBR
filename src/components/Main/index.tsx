import Button from 'components/Button'

import * as S from './styles'

export type MainProps = {
  jobs: string
}

const Main = ({ jobs }: MainProps) => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.png"
      alt="Imagem de um círculo com as cores da bandeira do Brasil."
    />
    <S.Title>Espaço para divulgação de vagas para front-enders.</S.Title>

    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Um desenvolvedor de frente para uma tela com código."
    />

    <Button as="a" href="/jobs">
      Ver as {jobs} vagas
    </Button>
  </S.Wrapper>
)

export default Main
