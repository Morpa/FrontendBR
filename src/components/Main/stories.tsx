import { Story, Meta } from '@storybook/react/types-6-0'
import Main, { MainProps } from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'title default',
    jobs: '344'
  }
} as Meta

export const Default: Story<MainProps> = (args) => <Main {...args} />
