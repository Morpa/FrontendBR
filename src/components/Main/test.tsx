import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Main from '.'

const props = {
  title: 'title default',
  jobs: '344'
}

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Main {...props} />)

    expect(
      screen.getByRole('heading', {
        name: /title default/i
      })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /344/i })).toHaveAttribute(
      'href',
      '/jobs'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
