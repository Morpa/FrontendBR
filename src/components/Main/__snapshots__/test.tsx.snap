// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  color: #FAFAFA;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  padding: 3.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .c4 {
  margin-top: 3.2rem;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
  text-align: center;
}

.c3 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

@media (max-width:768px) {
  .c2 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <img
    alt="Imagem de um círculo com as cores da bandeira do Brasil."
    class="c1"
    src="/img/logo.png"
  />
  <h1
    class="c2"
  >
    Espaço para divulgação de vagas para front-enders.
  </h1>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="c3"
    src="/img/hero-illustration.svg"
  />
  <a
    class="c4 c5"
    href="/jobs"
  >
    <span>
      Ver as 
      344
       vagas
    </span>
  </a>
</main>
`;
